### ISSUE TITLE
Add class prediction methods to InferenceModel

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#344]
Basically something like sklearn KNeighborsClassifier. We could also add nearest-centroid classification and others.

See #342 and #343
